### DISCUSSION TITLE
i cant push changes because of an old file bigger than 100mb

### DISCUSSION THREAD

**Discussion by kailingma** [community/community#51693] [Codespaces]
### Select Topic Area

Question

### Body

I was trying to upload a big file, but github stopped me, so i installed LFS but when i click sync changes it doesn't let me because of a large file i was trying to upload, then i deleted it in desperation, but it still wont sync

**Comment by airtower-luna** [comment]
Installing LFS won't magically make a large file in the history work, you'll have to rewrite the repository history to either move the file to LFS, or remove it completely (if you don't care about the old version any more). Mind that everyone who already has the old repository history will need to manually accept the changed history.

The process for removing a large file is basically the same as for an accidentally committed secret file: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository

If you want to learn about changing repository history in general, see: [Git Tools - Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)

Make sure to do a backup first if you care about actually keeping the file.

**Comment by valerio-so** [comment]
[ok]()

**Comment by kailingma** [comment]
how do i get the .git file for my repository as it says in BFG

**Comment by airtower-luna** [comment]
@kailingma, what exactly does BFG say? `.git` is a hidden folder inside your repository which you should not need to touch directly.

**Comment by kailingma** [comment]
[https://imgur.com/a/bjIBFnY](url) <<=== this

**Comment by kailingma** [comment]
thats part of the BFG docs [https://rtyley.github.io/bfg-repo-cleaner/](url)

**Comment by airtower-luna** [comment]
There's no mention of a `.git` file. What it tells you is to clone your repo into a local *bare* (no working tree) repository to work with. The directory name of bare repositories usually ends with `.git`. That's all.

**Comment by kailingma** [comment]
so how do i use it then?
